repository/src/main/resources/alfresco/action-services-context.xml

```diff
@@ -768,5 +769,22 @@
             <ref bean="TemplateService"/>
         </property>
     </bean>
+
+    <!-- Action Restriction Beans -->
+
+    <bean id="actionAccessRestrictionBase" abstract="true" class="org.alfresco.repo.action.access.ActionAccessRestrictionAbstractBase">
+        <property name="nodeService">
+            <ref bean="NodeService"/>
```

Comment:
Shouldn't we use a lower case `nodeService` here? Right now the only restriction is for admin users so it will work but if we think of it as a generic framework then I think security interceptors on the `NodeService` might prevent us from the real check.

Reply:
I don't know about it, the upper case NodeService is fine for now and should be fine for the current uses. If we/someone else meets a problem here, there's no issue with it being replaced to lowercase nodeService. The other way around might be an issue though (as we've had experience with during action security analysis).

I could theoretically set it as "nodeService" and make the variable private in ActionAccessRestrictionAbstractBase

Reply:
I don't know if it makes sense bu the scenario I had in mind is as follows:
1. As a regular user I change some node I have access to.
2. Some rule matches my node and an action is triggered.
3. I don't have access to the Action/Rule node created by admin.
4. It fails on checking `isActionCausedByRule` even without reaching the `innerVerifyAccessRestriction`

Reply:
I'll try to execute a scenario locally
Admin user creates a folder in his own space with rule of secured action,
Admin user creates a folder in shared location with rule of move action,
Regular user uploads file in shared location.

Theoretically that, if possible, could trigger the scenario. I don't know ACS that well though.

Reply:
It seems it has never been considered as a use case to be honest. I've tried to set up the mentioned scenario and didn't even finish triggering Copy action before getting "Access Denied.  You do not have the appropriate permissions to perform this operation.". Since I didn't get to finish copy action by not having access to the folder specified in a rule by admin, then I can't trigger other actions there regardless of the security interceptor being present or not on a restricted access action.

Reply:
I've tested same scenario when user has read access to both folder nodes, seems to work properly in that case.

Reply:
If it's not a case then just leave it as it is. As you wrote we can always change it in the future.